src/plugins/lua/known_senders.lua

```diff
@@ -197,6 +210,57 @@ local function known_senders_callback(task)
   end
 end
 
+local function check_known_incoming_mail_callback(task)
+  local sender = task:get_reply_sender()
+  if not sender then
+    lua_util.debugm(N, task, 'Could not get sender')
+    return nil
+  end
+
+  -- making sender key
+  lua_util.debugm(N, task, 'Sender: %s', sender)
+
+  local sender_string = lua_util.maybe_obfuscate_string(tostring(sender), settings, settings.sender_prefix)
+  local sender_key = make_key_replies(sender_string:lower(), 8)
+
+  lua_util.debugm(N, task, 'Sender key: %s', sender_key)
+
+  local list_of_senders = {}
+
+  local function redis_zrange_cb(err, data)
+    if err ~= nil then
+      rspamd_logger.errx(task, 'Couldn\'t get data from global replies set. Ended with error: %s', err)
+      return
+    end
+
+    list_of_senders = data
+    rspamd_logger.infox(task, 'Successfully got list: %s of verified senders', data)
+
+    lua_util.debugm(N, task, 'List of senders: %s', list_of_senders)
+
+    -- searching for sender in global replies set
+    if list_of_senders then
+      for _, sndr in ipairs(list_of_senders) do
+        if sndr == sender then
```

Comment:
In Lua, this comparison is O(1), so this code makes sense. In other languages, it is better to convert `list_of_senders` to a set and check that value in set should the length of `list_of_senders` > around 50-100 elements in general. Not a change request, just FYI.